pennylane/devices/default_mixed.py

```diff
@@ -0,0 +1,145 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+The default.mixed device is PennyLane's standard qubit simulator for mixed-state computations.
+
+It implements the necessary :class:`~pennylane._device.Device` methods as well as some built-in
+:mod:`qubit operations <pennylane.ops.qubit>`, providing a simple mixed-state simulation of
+qubit-based quantum circuits.
+"""
+
+import numpy as np
+
+from pennylane import QubitDevice
+from pennylane.operation import DiagonalOperation, Channel
+
+
+class DefaultMixed(QubitDevice):
+    """Default qubit device for performing mixed-state computations in PennyLane.
+
+    Args:
+        wires (int, Iterable[Number, str]): Number of subsystems represented by the device,
+            or iterable that contains unique labels for the subsystems as numbers
+            (i.e., ``[-1, 0, 2]``) or strings (``['ancilla', 'q1', 'q2']``).
+        shots (int): Number of times the circuit should be evaluated (or sampled) to estimate
+            the expectation values. Defaults to 1000 if not specified.
+            If ``analytic == True``, the number of shots is ignored
+            in the calculation of expectation values and variances, and only controls the number
+            of samples returned by ``sample``.
+        analytic (bool): indicates if the device should calculate expectations
+            and variances analytically.
+    """
+
+    name = "Default mixed-state qubit PennyLane plugin"
+    short_name = "default.mixed"
+
+    operations = {
```

Comment:
@ixfoduap, are all of these operations supported as of this PR? If not, I would suggest commenting out the ones that are still to be implemented.

Reply:
To the best of my knowledge, yes, they are all supported

Reply:
I suppose though, technically, _none_ of them are supported until the `apply()` method is implemented?

Reply:
Correct, technically not supported without `apply()`. Do you recommend removing them for now?

Reply:
We still want to fix tests before merging in right? Currently the tests fail because `apply()` is not defined - maybe we could do `apply = lambda *args, **kwargs: None` as suggested?

Reply:
> Correct, technically not supported without apply(). Do you recommend removing them for now?

Perhaps just commenting them out? The plugin tests,

```python
pl-device-test --device default.qubit --shots 10000 --analytic False
```

use this attribute to work out which tests to skip. So if an operation is listed here that currently does not 'work' (it doesn't produce the correct output probabilities), these tests will not be automatically skipped, and it will be hard to use the plugin device test during development.

But this is a relatively minor point.

Reply:
dummy `apply()` method now added. I'll check if any other test is failing and fix it.